### DISCUSSION TITLE
Without `cache: false` webpack detect changes only once

### DISCUSSION THREAD

**Discussion by dehghani-mehdi** [webpack/webpack#14398] [Q&A]
# Bug report

**What is the current behavior?**
webpack detect changes only once.

**If the current behavior is a bug, please provide the steps to reproduce.**

1. `git clone https://github.com/dehghani-mehdi/webpack-testcase.git`
2. `npm start`
3. Try to make any change, for example add a number to `<div>Welcome to React App</div>` inside `index,js`

You will see your change only the first time, after that webpack doesn't detect changes.

**What is the expected behavior?**
After any change, webpack should be able to detect it.

Adding `cache: false` resolve the issue, but I think it's a bug. I also tried ` cache: { type: 'filesystem' }`, but doesn't work.

**Other relevant information:**
webpack version:  `5.56.0`
Node.js version: `12.18.2`
Operating System: Windows 10 x64
Additional tools: -

**Comment by vankop** [comment]
Could you create reproducible repo? You probably using `webpack-dev-server`, so from this description your setup is unclear to me.

**Reply by dehghani-mehdi** [reply]
@vankop here you go: https://github.com/dehghani-mehdi/webpack-testcase (also first post updated)

**Reply by alexander-akait** [reply]
Change `import App from "./App";` on `import App from "./app";`

**Reply by dehghani-mehdi** [reply]
@alexander-akait It works, Could you explain why?

**Reply by alexander-akait** [reply]
Due case insensitivity logic for fs, webpack try to watch `App.js` file using Node.js API, but because your fs is case insensitive, Node.js is watching `app.js`, but inside webpack module have name `App.js` and logic for reloading is not working, I recommend to do setup linter and avoid rely on case insensitivity ability, it is not safe for many other places.
